### PR TITLE
Add support for UnusedMethodCall

### PR DESCRIPTION
**Pull request by aandresen-slack**
Add support for `UnusedMethodCall` to the Hakana `MustUse` attribute.

This required moving over `get_methodlike_id_from_call` that we implemented for the user migration.

### CODE REVIEW

File: src/analyzer/expr/expression_identifier.rs [hunk 1]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
...
+    call_expr: &oxidized::ast::CallExpr,
+    interner: &Interner,
+    resolved_names: &FxHashMap<u32, StrId>,
+    expr_types: &FxHashMap<(u32, u32), Rc<TUnion>>,
+) -> Option<FunctionLikeIdentifier> {
+    get_functionlike_id_from_call(call_expr, Some(interner), resolved_names)
+        .or_else(|| get_methodlike_id_from_call(call_expr, interner, expr_types))
+}
+
+pub fn get_methodlike_id_from_call(
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:232]: Prefer `get_method_id_from_call` here

File: src/analyzer/expr/expression_identifier.rs [hunk 2]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
...
+                lhs_expr.1.end_offset() as u32,
+            )) {
+                let t = lhs_expr_type.types.first().unwrap();
+                if let TAtomic::TNamedObject { name, .. } = t {
+                    Some(name)
+                } else {
+                    None
+                }
+            } else {
+                None
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:258]: This should be a `return None;`

File: src/analyzer/expr/expression_identifier.rs [hunk 3]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
...
+
+            let class_id = if let Some(lhs_expr_type) = expr_types.get(&(
+                lhs_expr.1.start_offset() as u32,
+                lhs_expr.1.end_offset() as u32,
+            )) {
+                let t = lhs_expr_type.types.first().unwrap();
+                if let TAtomic::TNamedObject { name, .. } = t {
+                    Some(name)
+                } else {
+                    None
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:255]: This should be a `return None;`

File: src/analyzer/expr/expression_identifier.rs [hunk 4]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
...
+                return None;
+            }
+
+            let class_id = if let Some(lhs_expr_type) = expr_types.get(&(
+                lhs_expr.1.start_offset() as u32,
+                lhs_expr.1.end_offset() as u32,
+            )) {
+                let t = lhs_expr_type.types.first().unwrap();
+                if let TAtomic::TNamedObject { name, .. } = t {
+                    Some(name)
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:253]: This should just be `name`

File: src/analyzer/expr/expression_identifier.rs [hunk 5]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
...
+                    None
+                }
+            } else {
+                None
+            };
+
+            if let aast::Expr_::Id(method_name_node) = &rhs_expr.2 {
+                if let Some(method_id) = interner.get(&method_name_node.1) {
+                    return Some(FunctionLikeIdentifier::Method(
+                        *class_id.unwrap_or(&StrId::EMPTY),
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:264]: With above changes you can get rid of the `.unwrap_or(...)`

File: src/analyzer/expr/expression_identifier.rs [hunk 6]
```diff
@@ -213,6 +219,62 @@ pub fn get_functionlike_id_from_call(
     }
 }
 
+pub fn get_id_from_call(
```
Comment (muglug) [inline src/analyzer/expr/expression_identifier.rs:222]: I'd prefer making this `get_functionlike_id_from_call` and then migrating the existing `get_functionlike_id_from_call` to `get_static_functionlike_id_from_call`

Comment (aandresen-slack) [reply src/analyzer/expr/expression_identifier.rs]: Ok updated all these. I wasn't sure what callers to move over but let me know what you think

**Review by muglug — APPROVED** [review]

**Comment by muglug** [comment]
Thanks!

**Pull request merged**